### PR TITLE
[11.0][IMP] stock_move_location: Allow move products from one location to another directly from selected quants

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa
This PR adds various improvements to this module:

1.  When in the wizard the user changes the location_destination_id all lines are erased, so the user loses the quantities to transfer to another location. With this PR only change the location_destination_id for all lines.

2. Add improvements that allow to do a transfer selecting quants directly from various origin locations. In this case you can not change origin location in the wizard. I try to no update the original behavior.

Thanks

### CODE REVIEW

File: stock_move_location/README.rst [hunk 1]
```diff
@@ -45,6 +46,7 @@ Usage
   it will move only the available quantity at the button press
 * Products will be moved and a form view of picking that did that will show up
 * If "PLANNED TRANSFER" is used - the picking won't be validated automatically
+* In quants view you has an action to move selected quants to another location
```
Comment (pedrobaeza) [inline stock_move_location/README.rst]: Please put as a different  paragraph in the section:
```rst
* If "PLANNED...

If you want to transfer a full quant:

#. Go to ...
```

File: stock_move_location/data/stock_move_location.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (pedrobaeza) [inline stock_move_location/data/stock_move_location.xml]: This should be named `stock_quant_views.xml`

**Review by pedrobaeza — APPROVED** [review]
Little details

**Review by victor-ditay — APPROVED** [review]
just the @pedrobaeza review others LGTM

**Comment by sergio-teruel** [comment]
@pedrobaeza @victormmtorres Changes done!!

File: stock_move_location/wizard/stock_move_location.py [hunk 3]
```diff
@@ -30,19 +31,38 @@ class StockMoveLocationWizard(models.TransientModel):
...
+        res['stock_move_location_line_ids'] = [(0, 0, {
+            'product_id': quant.product_id.id,
+            'move_quantity': quant.quantity,
+            'max_quantity': quant.quantity,
+            'origin_location_id': quant.location_id.id,
+            'lot_id': quant.lot_id.id,
+            'product_uom_id': quant.product_uom_id.id,
+            'custom': False,
+        }) for quant in quants]
+        res['origin_location_id'] = quants[:1].location_id.id
```
Comment (rousseldenis) [inline stock_move_location/wizard/stock_move_location.py]: Prefer using 'first' method:
```suggestion
        from odoo.fields import first
        res['origin_location_id'] = first(quants).location_id.id
```

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Minor change

File: stock_move_location/wizard/stock_move_location.py (hunk 3 above)
Comment (sergio-teruel) [reply stock_move_location/wizard/stock_move_location.py]: Hi Denis, I didn't know this function, but... ¿Can you explain me in this case why is better use first(with prefetch) than directly records[:1]?
I have done a profile for two cases and I do not see a big difference, even records[:1] is faster.. :smile: 
You have to take into account that I already access to records so the data is in cache.

**Profiler with records[:1]**
calls     queries   ms
1         0         0.02          @profile
                                  @api.model
                                  def default_get(self, fields):
1         1         1.46              res = super().default_get(fields)
1         0         0.01              if self.env.context.get('active_model', False) != 'stock.quant':
                                          return res
                                      # Load data directly from quants
1         0         0.03              quants = self.env['stock.quant'].browse(
1         0         0.02                  self.env.context.get('active_ids', False))
1         0         0.0               res['stock_move_location_line_ids'] = [(0, 0, {
                                          'product_id': quant.product_id.id,
                                          'move_quantity': quant.quantity,
                                          'max_quantity': quant.quantity,
                                          'origin_location_id': quant.location_id.id,
                                          'lot_id': quant.lot_id.id,
                                          'product_uom_id': quant.product_uom_id.id,
                                          'custom': False,
1         6         70.74             }) for quant in quants]
**_1         0         0.04              res['origin_location_id'] = quants[:1].location_id.id_**
1         0         0.0               return res

Total:
1         7         72.32   

**Profiler with method firts**
calls     queries   ms
1         0         0.01          @profile
                                  @api.model
                                  def default_get(self, fields):
1         1         1.25              res = super().default_get(fields)
1         0         0.01              if self.env.context.get('active_model', False) != 'stock.quant':
                                          return res
                                      # Load data directly from quants
1         0         0.02              quants = self.env['stock.quant'].browse(
1         0         0.02                  self.env.context.get('active_ids', False))
1         0         0.0               res['stock_move_location_line_ids'] = [(0, 0, {
                                          'product_id': quant.product_id.id,
                                          'move_quantity': quant.quantity,
                                          'max_quantity': quant.quantity,
                                          'origin_location_id': quant.location_id.id,
                                          'lot_id': quant.lot_id.id,
                                          'product_uom_id': quant.product_uom_id.id,
                                          'custom': False,
1         6         70.37             }) for quant in quants]
**_1         0         0.05              res['origin_location_id'] = first(quants).location_id.id_**
1         0         0.0               return res

Total:
1         7         71.74

Comment (rousseldenis) [reply stock_move_location/wizard/stock_move_location.py]: I agree, but as Odoo provides the helper method, it's safer to use it than indexes (and more logic as indexes are for lists not really for recordsets).

Comment (sergio-teruel) [reply stock_move_location/wizard/stock_move_location.py]: Ok, Thanks!! :smile:

Comment (pedrobaeza) [reply stock_move_location/wizard/stock_move_location.py]: @rousseldenis not really, as that notation is Python standard, and recordsets adopt that standards (the same as set logic operations).

**Comment by sergio-teruel** [comment]
@rousseldenis change done!

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by HviorForgeFlow — APPROVED** [review]
Functional review :+1: Nice feature :smile:

**Comment by HviorForgeFlow** [comment]
@sergio-teruel I see this feature is available for Inventory Users, it could be dangerous that a large amount of users in company could move freely quants between location without restriction. 

This is not a blocking request, but creating a group to enable this feature could be a good practice.

**Comment by pedrobaeza** [comment]
@hveficent IMO, a warehouse user is already able to do the same, but manually, so adding more layers is overkilling. This can't be done accidentally, as you have to accept the wizard popup, so for me it's OK, although other opinions are welcome.

**Comment by rousseldenis** [comment]
> @hveficent IMO, a warehouse user is already able to do the same, but manually, so adding more layers is overkilling. This can't be done accidentally, as you have to accept the wizard popup, so for me it's OK, although other opinions are welcome.

Yes, but it's an open door to massive changes. I agree with @hveficent, but the inventory user could be added by default (and let admin change it if required)

**Comment by pedrobaeza** [comment]
Then it's OK as is?

**Comment by sergio-teruel** [comment]
Ok, I have added the stock.group_stock_user group to quants action

**Pull request merged**